lang/src/org/partiql/lang/util/UtilityExtensions.kt

```diff
@@ -1,25 +1,34 @@
-/*
- * Copyright 2019 Amazon.com, Inc. or its affiliates.  All rights reserved.
- *
- * Licensed under the Apache License, Version 2.0 (the "License").
- *  You may not use this file except in compliance with the License.
- * A copy of the License is located at:
- *
- *      http://aws.amazon.com/apache2.0/
- *
- *  or in the "license" file accompanying this file. This file is distributed on an "AS IS" BASIS,
- *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific
- *  language governing permissions and limitations under the License.
- */
-
 package org.partiql.lang.util
 
+import org.partiql.lang.eval.builtins.Time.Companion.MAX_PRECISION_FOR_TIME
+import org.partiql.lang.eval.builtins.Time.Companion.SECONDS_PER_HOUR
+import org.partiql.lang.eval.builtins.Time.Companion.SECONDS_PER_MINUTE
+import java.time.LocalTime
 import java.time.ZoneOffset
 import kotlin.math.absoluteValue
 
 /**
  * Returns the string representation of the [ZoneOffset] in HH:mm format.
  */
-fun ZoneOffset.getOffsetHHmm(): String {
-    return "${if(totalSeconds >= 0) "+" else "-"}${(totalSeconds / 3600).absoluteValue.toString().padStart(2, '0')}:${((totalSeconds / 60) % 60).absoluteValue.toString().padStart(2, '0')}"
-}
+fun ZoneOffset.getOffsetHHmm(): String =
+    (if(totalSeconds >= 0) "+" else "-") +
+    (totalSeconds / SECONDS_PER_HOUR).absoluteValue.toString().padStart(2, '0') +
+    ":" +
+    ((totalSeconds / SECONDS_PER_MINUTE) % SECONDS_PER_MINUTE).absoluteValue.toString().padStart(2, '0')
+
+/**
+ * Gets the number of most significant digits (excluding trailing zeros) in the fractional part of the second represented by [nano] of [LocalTime]
+ * For example, localtime with nano = 123000000 returns precision as 3
+ */
+fun LocalTime.getPrecision(): Int = when (nano) {
+    0 -> 0
+    else -> {
+        var trailingZeros = 0
+        var nanos = nano
+        while (nanos % 10 == 0) {
+            trailingZeros ++
+            nanos /= 10
+        }
+        MAX_PRECISION_FOR_TIME - trailingZeros
+    }
+}
```

Comment:
I don't think this is right.  Given some nanoseconds, you cannot infer precision.  The only way to infer precision is from the textual representation otherwise it must be given to us.

Reply:
Removed this method.